language/move-prover/src/spec_translator.rs

```diff
@@ -618,6 +620,27 @@ impl<'env> SpecTranslator<'env> {
             distribution.add(self, for_entry_point, cond);
         }
 
+        // emit preconditions for modifies
+        for (ty, targets) in func_target.get_modify_targets() {
```

Comment:
So do we really need to emit these for indirect calls?

Consider f calls g calls h. g and h have modifies clauses. We already proven that M_g subsumes M_h. We do not need to prove it again when we verify f.

This would call for adding these conditions to the ConditionDistribution in the regular way. However, we would need to also generate different function bodies for the indirect and the direct call, where the code only in the direct case need to make the assertions on memory update. I thought doing  something similar for global invariants.

I'm fine we would do this (if my argument is correct) in a future PR, but then please add a TODO.